### PR TITLE
chore: update build targets

### PR DESCRIPTION
**Pull request by Uzlopak**

### CODE REVIEW

**Comment by Uzlopak** [comment]
@wolfy1339 

should we target a beta branch?

**Comment by wolfy1339** [comment]
Do we want to push other breaking changes? Or changes that can only happen with this PR merged?

**Comment by Uzlopak** [comment]
We gonna break things for esm only

**Comment by wolfy1339** [comment]
I'd say yes let's target a beta branch

**Comment by Uzlopak** [comment]
@wolfy1339 

PTAL

**Comment by wolfy1339** [comment]
Update the package.json engines field as well

**Comment by Uzlopak** [comment]
@wolfy1339 
done ;). also modified the ci.yml

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,39 +1,118 @@
...
+        with:
+          languages: javascript-typescript
+          queries: security-and-quality
+
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v3
+        with:
+          category: "/language:javascript-typescript"
+
+  lint:
```
Comment (wolfy1339) [inline .github/workflows/test.yml:72]: There needs to be a wrapper job so we don't require specific node versions I'm branch protection rules.

Comment (Uzlopak) [reply .github/workflows/test.yml:72]: what do you mean?

Comment (wolfy1339) [reply .github/workflows/test.yml:72]: Like this
https://github.com/probot/probot/blob/880f2b93160e88bb9f5cdbd336c2b68702aa51f1/.github/workflows/test.yml#L118-L122

Comment (Uzlopak) [reply .github/workflows/test.yml:72]: I added the wrapper, but i dont see the point

**Comment by Uzlopak** [comment]
It broke now...i am now on the way home. Any idea?

**Pull request merged**